WooCommerce/Classes/Authentication/Store Creation/Profiler/RequiredStoreCreationProfilerQuestionView.swift

```diff
@@ -0,0 +1,90 @@
+import Combine
+import SwiftUI
+
+/// Handles the navigation action and provides the continue button state in a required profiler question view during store creation.
+/// The question is not skippable.
+protocol RequiredStoreCreationProfilerQuestionViewModel {
+    /// Called when the continue button is tapped.
+    func continueButtonTapped() async
+
+    /// Whether the continue button is enabled for the user to continue.
+    var isContinueButtonEnabled: AnyPublisher<Bool, Never> { get }
+}
+
+/// Shows a mandatory profiler question during the store creation flow with a generic content.
+struct RequiredStoreCreationProfilerQuestionView<QuestionContent: View>: View {
+    private let viewModel: StoreCreationProfilerQuestionViewModel & RequiredStoreCreationProfilerQuestionViewModel
+    @ViewBuilder private let questionContent: () -> QuestionContent
+    @State private var isWaitingForCompletion: Bool = false
+    @State private var isContinueButtonEnabled: Bool = false
+
+    init(viewModel: StoreCreationProfilerQuestionViewModel & RequiredStoreCreationProfilerQuestionViewModel,
+         @ViewBuilder questionContent: @escaping () -> QuestionContent) {
+        self.viewModel = viewModel
+        self.questionContent = questionContent
+    }
+
+    var body: some View {
+        ScrollView {
+            StoreCreationProfilerQuestionView<QuestionContent>(viewModel: viewModel, questionContent: questionContent)
+        }
+        .safeAreaInset(edge: .bottom) {
+            VStack {
+                Divider()
+                    .frame(height: Layout.dividerHeight)
+                    .foregroundColor(Color(.separator))
+                Button(Localization.continueButtonTitle) {
+                    Task { @MainActor in
+                        isWaitingForCompletion = true
+                        await viewModel.continueButtonTapped()
+                        isWaitingForCompletion = false
+                    }
+                }
+                .buttonStyle(PrimaryLoadingButtonStyle(isLoading: isWaitingForCompletion))
+                .disabled(!isContinueButtonEnabled)
+                .padding(Layout.defaultPadding)
+            }
+            .background(Color(.systemBackground))
+        }
+        // Disables large title to avoid a large gap below the navigation bar.
+        .navigationBarTitleDisplayMode(.inline)
+        .onReceive(viewModel.isContinueButtonEnabled) { isContinueButtonEnabled in
+            self.isContinueButtonEnabled = isContinueButtonEnabled
```

Comment:
❓ I don't understand why we have a local var `isContinueButtonEnabled` instead of using the one from the `viewModel`?

Reply:
The view model's `isContinueButtonEnabled` is of type `AnyPublisher<Bool, Never>` since we can't have a `@Published` variable in the protocol. That's why it's manually bound to a state variable so that the UI can respond to changes